File: core/java/src/main/java/hara/lang/protocol/IExInfo.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.iexinfo", name = "IExInfo")
public interface IExInfo {
  @HaraMethod(value = "data", arity = 1)
  public IMetadata getData();
}

File: core/rust/src/lang/protocol/iexinfo.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.iexinfo", name = "IExInfo")]
pub trait IExInfo {
    type Data;

    #[hara_method(value = "data", arity = 1)]
    fn data(&self) -> Self::Data;
}
